lib/matplotlib/tests/test_dates.py

```diff
@@ -22,6 +23,52 @@
 import matplotlib.dates as mdates
 
 
+def test_date_numpyx():
```

Comment:
any reason this isn't done with base/time/timenp being pytest fixtures and the test parameterized or split into two?

Reply:
Because I can understand this, and I have no idea what a pytest fixture is.  Is there an advantage to doing something more obscure than a simple `assert`?

Reply:
Only cause you're repeating yourself a lot so it's hard to tell what the specific tests are. Factoring out common features into a fixture means you can make two tests that are uniquely named for the things they're testing.

Reply:
I'll look into it.  OTOH, I find some of the tests aren't very easy to follow.  Sometime verbosity is a good thing!

Reply:
OK, happy to take suggestions, but I find the uses of fixtures in the existing tests makes for pretty obfuscated code.

Reply:
*shrugs* can agree on that somewhat, either way it's not something I'd hold the PR up on.

Reply:
Well, I didn't mean to say I wasn't willing to learn, just that I don't get it...  More than happy if you wanted to push a change or give me some pseudo code to get it to work...  Thanks!

Reply:
Sorry for missing your reply (the embedded ones seem to not pop up as notifications), and will totally try to remember your suggestion for the future.  I should have been way clearer on what I wanted and given you an example. :/